Code/src/main/java/nl/utwente/group10/ui/CustomUIPane.java

```diff
@@ -32,8 +32,8 @@ public CustomUIPane() {
      */
     public final void invalidate() {
         for (Node node : getChildren()) {
-            if (node instanceof DisplayBlock) {
-                ((DisplayBlock)node).invalidate();
+            if (node instanceof Block) {
+                ((Block)node).invalidate();
```

Comment:
Ik zie dat je het document inmiddels aangepast hebt, als ik me niet vergist stond in het document eerst beschreven dat .invalidate() aanroepen op de pane ALLE **Blocks** zou moeten re-evaluaten. Dit was echter in de code niet het geval, ik had echter beter kunnen weten door de java doc te lezen en op dat moment aan de bel te trekken. 

Ik kwam destijds echter tot een andere conclusie: FunctionBlocks geven indirect OOK een "status" weer ook als ze NIET verbonden zijn met een DisplayBlock, hoe kun je anders aangeven dat er missing arguments zijn of dat er een type mismatch is? Mij leek het dat op het moment dat er ERGENS in een tree de status van een Block verandert dat die gehele tree opnieuw ge-evaluate moet worden. Dit dus ook op het moment als er connecties van en naar onder andere een FunctieBlock gemaakt of verbroken worden. De gedachte dat dit een betere aanpak zou zijn werd verder versterkt door je eerdere opmerking over het gebruik van asExpr(), waarom die aanroepen als je niets met het resultaat doet? Daar had je gelijk in. Die aanpak is niet netjes maar er moet toch een manier zijn om een willekeurig type Block zich te laten re-evaluaten (context afhankelijk per Block implementatie). Waarom dan niet .invalidate() hiervoor gebruiken? het leek namelijk alsof dit exact het gedrag was dat de methode vertoonde voor DisplayBlock.

Hoe dan ook blijkt dat dit toch een pittig discussie punt is waarvan de oplossing nog niet duidelijk is. We doen er dan ook goed aan om hier liever eerder dan later hierna te kijken en proberen er een aanpak voor te verzinnen.

Het document suggereert om elke Block een Haskell expressie te laten bijhouden (iets wat ik eerder in gedachten had) maar haalt dit niet het hele principe van de eerder gemaakte boom structuur weg?